### ISSUE TITLE
e2e (and some integration) test Makefile targets are not actually running tests

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#3331]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running e2e tests e.g. as part of a release (example https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3846157749/jobs/6553117609#step:7:60) one can observe that it only deploys a cluster and no actual tests are being run, exemplar logs:

```
INFO: no cluster name provided, using generated name kic-d6d127a8-39db-484b-b685-3182f3e5e028
INFO: validating cluster version requirements
INFO: configuring the GKE cluster NAME=(kic-d6d127a8-39db-484b-b685-3182f3e5e028) VERSION=(v1.21) PROJECT=(***) LOCATION=(***)
INFO: building cluster kic-d6d127a8-39db-484b-b685-3182f3e5e028 (this can take some time)
INFO: verifying that the cluster can be communicated with
INFO: server version found: v1.21.14-gke.8500
```

That is due to the fact how the makefile commands are structured e.g. in https://github.com/Kong/kubernetes-ingress-controller/blob/996bc14da54409115b263ff1092c41b3667a4294/Makefile#L369-L380

mark that this only runs `KUBERNETES_CLUSTER_NAME="${CLUSTER_NAME}" go run hack/e2e/cluster/deploy/main.go` and the rest is just an argument list to this script.

Additionally, given the recent problems with GKE limits #3326, it's also worth noting that because of the above we're not doing [the cleanup that we intended](https://github.com/Kong/kubernetes-ingress-controller/blob/996bc14da54409115b263ff1092c41b3667a4294/Makefile#L355) and created clusters have to either be deleted manually in GKE console or wait for the cleanup job's cron https://github.com/Kong/kubernetes-ingress-controller/blob/e23071caec30346e623677b5dd8471e61b025f81/.github/workflows/cleanup.yaml#L5.

### Expected Behavior

Tests are actually run when triggered.

**Comment by pmalek** [comment]
After the fix in #3335 the e2e tests workflow is able to finish in time but no tests are actually being run, e.g. https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3852654346/jobs/6564968137#step:6:60.

I've made an attempt to setup a GKE cluster and reuse it in tests jobs running in parallel #3340 but that resulted in many different failures ( https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3856537022 ) mostly stemming from the fact that those tests deploy the same resources hence e.g. conflicts when `kubectl apply`ing manifests.

In such case I see the following possibilities:

- Run all those tests in a single workflow sharing the GKE cluster: this will be slow since because of the above limitation (tests cannot run in parallel) we'll have to run only 1 test at a time
- Run all tests in separate GKE clusters: wasteful but in runtime PoV it will be identical to above mentioned approach. The advantage of this is that we get complete isolation. The only thing we'll need to solve is #3326 since we still cannot create subnetwork during cluster creation via API. More details in #3326 .

**Comment by czeslavo** [comment]
It was fixed by https://github.com/Kong/kubernetes-ingress-controller/pull/3378.
